### ISSUE TITLE
Axios Response Typings Allow Incorrect Code

### ISSUE THREAD

**Issue by josephearl** [axios/axios#6105]
### Describe the bug

The following code should fail type checking but does not:

```ts
interface SomeBody {
  field1: string;
  field2: string;
}

async function fetchSomething(): Promise<SomeBody > {
  // Buggy code is missing a call to .data on the AxiosResponse
  return await axios.get('http://example.com/some/api'); // TypeScript is fine, but it should not be
}
```

If you introduce an intermediate variable typechecking does fail:

```ts
async function fetchSomething(): Promise< SomeBody > {
  const response = await axios.get('http://example.com/some/api');
  return response; // TS2322: Type  AxiosResponse<any, any>  is not assignable to type SomeBody
}
```

### To Reproduce

See https://codesandbox.io/p/sandbox/axios-ts-bug-gty2pg?file=%2Findex.ts%3A20%2C1

### Expected behavior

The following code fails type checking as it does when an intermediate variable is introduced:

```ts
interface SomeBody {
  field1: string;
  field2: string;
}

async function fetchSomething(): Promise<SomeBody > {
  // Buggy code is missing a call to .data on the AxiosResponse
  return await axios.get('http://example.com/some/api'); // TypeScript is fine, but it should not be
}
```

See https://github.com/microsoft/TypeScript/issues/56532?notification_referrer_id=NT_kwDOACZY-7I4NTM0MTk2OTc5OjI1MTMxNDc#issuecomment-1826791789 for a possible fix to the typings.

### Axios Version

1.6.2

### Browser

N/A

### Browser Version

N/A

### Node.js Version

18.18.2

### OS

MacOS

### Additional Library Versions

```bash
N/A
```

**Comment by Vayras** [comment]
Hi, @josephearl  I would like to contribute towards this issue, can you please assign me to this? thanks.

**Comment by Vayras** [comment]
@josephearl Thanks for letting me know , Will surely work on this issue

**Comment by NiharPhansalkar** [comment]
@josephearl Can we be sure this is a bug in Axios? I queried Bard on the issue. Let me know what you think.

Prompt 1: https://g.co/bard/share/f351e5b2eeec

Prompt 2: https://g.co/bard/share/a263ce7031ff

It does seem like this is how typescript behaves when encountering async functions.

**Comment by jasonsaayman** [comment]
i agree i think this is bit of a TS quirk but correctly functioning from a Axios POV
